evaluation/numpy/segmentation_and_tracking_quality.py

```diff
@@ -136,25 +177,19 @@ def update_state(self, y_true: np.ndarray, y_pred: np.ndarray, sequence_id=0):
                                 semantic_label, self._num_classes)
       semantic_prediction = np.where(semantic_prediction != self._ignore_label,
                                      semantic_prediction, self._num_classes)
+    # TODO(meijieru): weight the confusion matrix.
     if sequence_id in self._iou_confusion_matrix_per_sequence:
-      idxs = (np.reshape(semantic_label, [-1]) <<
```

Comment:
The original implementation contains shift operations, which significantly speeds up the computation for large datasets.
Is it possible to keep it?

Reply:
If we could make the assumption that the only possible weights are in `[0.5, 1]`, then we could also encode the weights when doing the `np.unique`.

Reply:
We could raise error if weights are not in [0.5, 1] and add comment that this assumption is need for this part of code. What do you think?